src/utils/useCUSDPrice.ts

```diff
@@ -59,23 +72,28 @@ export function useCUSDPrices(tokens?: Token[]): (Price | undefined)[] | undefin
 
 /**
  * Returns the price in cUSD of the input currency
- * @param currency currency to compute the cUSD price of
+ * @param token the token to get the cUSD price of
+ * @param totalSupplyOfStakingToken token's total lp supply
  */
-export function useCUSDPrice(token?: Token): Price | undefined {
+export function useCUSDPrice(token?: Token, totalSupplyOfStakingToken?: TokenAmount): Price | undefined {
   const {
     network: { chainId },
   } = useContractKit()
   const CUSD = cUSD[chainId as unknown as UbeswapChainId]
   const celo = CELO[chainId as unknown as UbeswapChainId]
+  const mcUSD = MCUSD[chainId as unknown as UbeswapChainId]
   const tokenPairs: [Token | undefined, Token | undefined][] = useMemo(
     () => [
       [token && currencyEquals(token, CUSD) ? undefined : token, CUSD],
       [token && currencyEquals(token, celo) ? undefined : token, celo],
+      [token && mcUSD && currencyEquals(token, mcUSD) ? undefined : token, mcUSD ? mcUSD : undefined],
       [celo, CUSD],
     ],
-    [CUSD, celo, token]
+    [CUSD, celo, mcUSD, token]
   )
-  const [[, cUSDPair], [, celoPair], [, celoCUSDPair]] = usePairs(tokenPairs)
+  const [[, cUSDPair], [, celoPair], [, mcUSDPair], [, celoCUSDPair]] = usePairs(tokenPairs)
```

Comment:
Is it possible to fetch the pool size from `usePairs`?

I'm not a fan of passing in `totalSupplyOfStakingToken` into a function as general as `useCUSDPrice`

Reply:
@BriungRi updated the PR to remove `totalSupplyOfStakingToken` optional param, now we're leveraging `usePairs`, let me know what you think!